src/jit/hwintrinsiclistxarch.h

```diff
@@ -630,6 +630,20 @@ HARDWARE_INTRINSIC(POPCNT_PopCount,                                 "PopCount",
 //  POPCNT Intrinsics
 HARDWARE_INTRINSIC(POPCNT_X64_PopCount,                             "PopCount",                                     POPCNT_X64,   -1,               0,           1,     {INS_invalid,           INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_popcnt,         INS_invalid,        INS_invalid},           HW_Category_Scalar,                 HW_Flag_NoFloatingPointUsed|HW_Flag_NoRMWSemantics|HW_Flag_MultiIns)
 
+// ***************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
+//                 Intrinsic ID                                     Function name                                   ISA         ival        SIMD size       NumArg                                                                                                     instructions                                                                                                     Category                            Flags
+//                                                                                                                                                                      {TYP_BYTE,              TYP_UBYTE,          TYP_SHORT,          TYP_USHORT,         TYP_INT,            TYP_UINT,           TYP_LONG,           TYP_ULONG,          TYP_FLOAT,          TYP_DOUBLE}
+// ***************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
+//  ADDCARRY Intrinsics
+HARDWARE_INTRINSIC(ADDCARRY_Add,                                    "AddWithCarry",                                 ADDCARRY,     -1,               0,           1,     {INS_invalid,           INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid,        INS_adc,            INS_invalid,        INS_invalid,        INS_invalid,        INS_invalid},           HW_Category_Scalar,                 HW_Flag_Commutative)
```

Comment:
This is likely going to require special codegen.

Reply:
This will likely also require special handling in lowering to ensure it handles the various cases correctly.

When dealing with a single `AddWithCarry`, you are going to need to generate something akin to:
```asm
add carry, -1
adc left, right
setc carry
```

And also optimize that when dealing with multiple sequential `AddWithCarry` so that you instead get something nicer like:
```asm
add carry, -1
adc left1, right1
adc left2, right2
setc carry
```

Reply:
> This will likely also require special handling in lowering to ensure it handles the various cases correctly.

Not likely. Definitely. These intrinsics are basically a pain in the back to support in the JIT. ATM I'm not even sure if it's actually possible to do this properly.